### ISSUE TITLE
Add a sandboxed process API locked by a permissions system - c.f. `child_process`

### ISSUE THREAD

**Issue by zm-cttae** [microsoft/vscode#174714]
We need a replacement to `child_process` that isn't an attack vector per #52116.
This could easily take the API shape of `atom.BufferedProcess`: https://github.com/atom/atom/blob/1c3bd35ce238dc0491def9e1780d04748d8e18af/src/buffered-node-process.js#L14-L34

Keywords: spawning, process, child_process, node.js, vscode API, terminals, problems, output, compiler, linter

**Comment by zm-cttae** [comment]
This should ideally be tabled until after #174715

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by zm-cttae** [comment]
I don't like this interpretation of 'agile'. Why is this ticket not being iceboxed? It explicitly depends on a permissions extension API and a corresponding trust UI.

**Comment by zm-cttae** [comment]
Closing as this proposal has failed. Blocking issue #174715 is due to be superseded by a more sound proposal.
